### ISSUE TITLE
Can't activate component in nested secondary outlet using RouterLink

### ISSUE THREAD

**Issue by refpolk** [angular/angular#28589]
# 🐞 bug report

### Affected Package

The issue is caused by package @angular/router.

### Is this a regression?

I don't know.

### Description

My application component tree is the following :

```
AppComponent
    Level0Component*
        Level1Component
        Aux1
        Aux2
            Level2Component*
                Level3Component
                Aux3
                Aux4
```

Level0Component and Level2Component both have a secondary router outlet (outlet1 and outlet2). I'm able to activate Level1Component in the primary outlet of Level0Component and Aux1Component or Aux2Component in it's secondary outlet outlet1 by using this syntax :

```
<a [routerLink]="['/level-0', { outlets: { primary: 'level-1', outlet1: 'aux-1' } }]">L1-A1</a>
<a [routerLink]="['/level-0', { outlets: { primary: 'level-1', outlet1: 'aux-2' } }]">L1-A2</a>
```

However I can't activate Level3Component in the primary outlet of Level2Component and Aux3Component or Aux4Component in it's secondary outlet outlet2 (while still activating Level1Component and Aux1Component or Aux2Component in Level0Component).

Below is the syntax I'm using. I get an `Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'level-0'` error when clicking those links.

```
<a [routerLink]="['/level-0', { outlets: { primary: ['level-1', { outlets: { primary: ['level-2', { outlets: { primary: ['level-3'], outlet2: 'aux-3' } }] } }], outlet1: 'aux-1' } }]">L1-A1/L2/L3-A3</a>
<a [routerLink]="['/level-0', { outlets: { primary: ['level-1', { outlets: { primary: ['level-2', { outlets: { primary: ['level-3'], outlet2: 'aux-4' } }] } }], outlet1: 'aux-1' } }]">L1-A1/L2/L3-A4</a>
```

## 🔬 Minimal Reproduction

https://stackblitz.com/fork/angular-nested-auxiliary-routes 

The problem can be reproduced by clicking the `L1-A1/L2/L3-A3` and `L1-A1/L2/L3-A4` links.

## 🔥 Exception or Error
<pre><code>Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'level-0'</code></pre>

Notice the `a/u/x/-/4` in the URL :
<pre><code>
Router Event: NavigationStart browser_adapter.ts:108:30
NavigationStart(id: 3, url: '/level-0/(level-1/(level-2/(level-3//outlet2:a/u/x/-/4))//outlet1:aux-1)') browser_adapter.ts:102:28
Object { id: 3, url: "/level-0/(level-1/(level-2/(level-3//outlet2:a/u/x/-/4))//outlet1:aux-1)", navigationTrigger: "imperative", restoredState: null }
browser_adapter.ts:102:28
Router Event: NavigationError browser_adapter.ts:108:30
NavigationError(id: 3, url: '/level-0/(level-1/(level-2/(level-3//outlet2:a/u/x/-/4))//outlet1:aux-1)', error: Error: Cannot match any routes. URL Segment: 'level-0') browser_adapter.ts:102:28
Object { id: 3, url: "/level-0/(level-1/(level-2/(level-3//outlet2:a/u/x/-/4))//outlet1:aux-1)", error: Error }
browser_adapter.ts:102:28
ERROR Error: "[object Object]"
	js 0.8.26/dist/zone.js:814:31
	js 0.8.26/dist/zone.js:771:17
	js 0.8.26/dist/zone.js:873:17
	js 0.8.26/dist/zone.js:421:17
	core 7.0.1/bundles/core.umd.js:14232:28
	js 0.8.26/dist/zone.js:420:17
	js 0.8.26/dist/zone.js:188:28
	js 0.8.26/dist/zone.js:595:25
	js 0.8.26/dist/zone.js:500:21
	js 0.8.26/dist/zone.js:1540:9
	js 0.8.26/dist/zone.js:1566:17
errors.ts:35:8
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.0.7
Node: 10.14.2
OS: win32 x64
Angular: 7.1.2
... common, core

Package                             Version
-------------------------------------------------------------
@angular-devkit/architect           0.10.7
@angular-devkit/build-angular       0.10.7
@angular-devkit/build-optimizer     0.10.7
@angular-devkit/build-webpack       0.10.7
@angular-devkit/core                7.0.7
@angular-devkit/schematics          7.0.7
@angular/cli                        7.0.7
@angular/compiler                   7.0.4
@angular/compiler-cli               7.0.4
@angular/forms                      7.0.4
@angular/language-service           7.0.4
@angular/platform-browser           7.1.1
@angular/platform-browser-dynamic   7.0.4
@angular/router                     7.1.1
@ngtools/webpack                    7.0.7
@schematics/angular                 7.0.7
@schematics/update                  0.10.7
rxjs                                6.3.3
typescript                          3.1.6
webpack                             4.19.1
</code></pre>

Thank you for your time in advance.

**Comment by lijiadashaoye** [comment]
你可以看一下这个：https://github.com/lijiadashaoye/nameOutlet.git

**Comment by refpolk** [comment]
@lijiadashaoye : I don't see any RouterLink in your sample, I'm not sure what you mean

**Comment by lijiadashaoye** [comment]
你疑问的是路由跳转以及路由的嵌套使用问题，没必要纠结使用routerlink还是navigate触发路由切换，希望你看懂里面的路由跳转思路

**Comment by lijiadashaoye** [comment]
> @lijiadashaoye：我的样本中没有看到任何RouterLink，我不确定你的意思

你疑问的是路由跳转以及路由的嵌套使用问题，没必要纠结使用routerlink还是navigate触发路由切换，希望你看懂里面的路由跳转思路

**Comment by lijiadashaoye** [comment]
> @lijiadashaoye：我的样本中没有看到任何RouterLink，我不确定你的意思

你疑问的是路由跳转以及路由的嵌套使用问题，没必要纠结使用routerlink还是navigate触发路由切换，希望你看懂里面的路由跳转思路

**Comment by atscott** [comment]
related to #13589

**Comment by atscott** [comment]
I believe this was partially fixed by 3af585c004a7785015d5a4af4640fc56365220ad

However, it does not fully work yet due to #18928. For the example 

```
['/level-0', { outlets: { primary: ['level-1', { outlets: { primary: ['level-2', { outlets: { primary: ['level-3'], outlet2: ['aux-3'] } }] } }], outlet1: 'aux-1' } }]
```

 if I change the deepest aux outlet command from `outlet2: 'aux-3'` to `outlet2: ['aux-3']`, the generated url will be 
`/level-0/(level-1/level-2/(level-3//outlet2:aux-3)//outlet1:aux-1)`

Closing as duplicate of #18928.
